assets/tests/components/lateView.test.tsx

```diff
@@ -282,66 +253,48 @@ describe("LateView", () => {
-    expectIsChecked(wrapper, lateMasterCheckboxSelector).toEqual(false)
-
-    act(() => {
-      wrapper
-        .find(
-          '.m-late-view__late-buses .m-late-view__data-row--unselected input[type="checkbox"]'
-        )
-        .first()
-        .simulate("click")
-    })
-    expectIsIndeterminate(wrapper, lateMasterCheckboxSelector).toEqual(true)
-    expectIsChecked(wrapper, lateMasterCheckboxSelector).toEqual(false)
-
-    act(() => {
-      wrapper
-        .find(
-          '.m-late-view__late-buses .m-late-view__data-row--unselected input[type="checkbox"]'
-        )
-        .last()
-        .simulate("click")
-    })
-    expectIsIndeterminate(wrapper, lateMasterCheckboxSelector).toEqual(false)
-    expectIsChecked(wrapper, lateMasterCheckboxSelector).toEqual(true)
-
-    expectIsIndeterminate(wrapper, missingMasterCheckboxSelector).toEqual(false)
-    expectIsChecked(wrapper, missingMasterCheckboxSelector).toEqual(false)
-
-    act(() => {
-      wrapper
-        .find(
-          '.m-late-view__missing-logons .m-late-view__data-row--unselected input[type="checkbox"]'
-        )
-        .first()
-        .simulate("click")
-    })
-    expectIsIndeterminate(wrapper, missingMasterCheckboxSelector).toEqual(true)
-    expectIsChecked(wrapper, missingMasterCheckboxSelector).toEqual(false)
-
-    act(() => {
-      wrapper
-        .find(
-          '.m-late-view__missing-logons .m-late-view__data-row--unselected input[type="checkbox"]'
-        )
-        .last()
-        .simulate("click")
-    })
-    expectIsIndeterminate(wrapper, missingMasterCheckboxSelector).toEqual(false)
-    expectIsChecked(wrapper, missingMasterCheckboxSelector).toEqual(true)
+    expectIndeterminateValue(result, lateMasterCheckboxTestId, false),
+      expectCheckedValue(result, lateMasterCheckboxTestId, false)
+
+    await userEvent.click(
+      result.getByTestId(`row-checkbox-${lateVehicle1.runId}`)
+    )
+    expectIndeterminateValue(result, lateMasterCheckboxTestId, true)
+    expectCheckedValue(result, lateMasterCheckboxTestId, false)
+
+    await userEvent.click(
+      result.getByTestId(`row-checkbox-${lateVehicle2.runId}`)
+    )
+
+    expectIndeterminateValue(result, lateMasterCheckboxTestId, false)
+    expectCheckedValue(result, lateMasterCheckboxTestId, true)
+
+    expectIndeterminateValue(result, missingMasterCheckboxId, false),
+      expectCheckedValue(result, missingMasterCheckboxId, false)
+
+    await userEvent.click(
+      result.getByTestId(`row-checkbox-${missingLogonGhost1.runId}`)
+    )
+    expectIndeterminateValue(result, missingMasterCheckboxId, true),
+      expectCheckedValue(result, missingMasterCheckboxId, false)
+
+    await userEvent.click(
+      result.getByTestId(`row-checkbox-${missingLogonGhost2.runId}`)
+    )
+
+    expectIndeterminateValue(result, missingMasterCheckboxId, false),
+      expectCheckedValue(result, missingMasterCheckboxId, true)
```

Comment:
This was already a long block of code in the original test, but some comments highlighting the individual interactions being tested at different spots would probably make it a lot more readable if you're able to add them (totally optional!).

Reply:
Yeah good point! I debated between breaking these tests up more, but commenting seems like a helpful change w/ a lower surface area